ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAlgorithm.scala

```diff
@@ -85,4 +85,20 @@ abstract class H2OAlgorithm[P <: Model.Parameters : ClassTag] extends Estimator[
   }
 
   override def copy(extra: ParamMap): this.type = defaultCopy(extra)
+
+  protected def createKey[T <: Keyed[T]](modelId: String): Key[T] = {
+    val key = Key.make[T](modelId)
+    if(H2O.containsKey(key)) {
+      var i = 0
+      var replacement = key
+      do {
```

Comment:
Just nit: Maybe we could expose the replacement functionality to a new, small private method and keep this method as:

if(keyExists){
 createReplacement()
} else {
key}